amethyst_renderer/src/pipe/stage.rs

```diff
@@ -69,29 +69,28 @@ impl Stage {
     /// Divides the given `Encoder`s into different sized chunks, pairs each
     /// chunk with a corresponding `Pass`, and returns the resulting batches as
     /// a `DrawUpdate`.
-    pub(crate) fn apply<'a, E>(&'a self, mut encoders: E, scene: &'a Scene) -> DrawUpdate<'a>
-        where E: Iterator<Item = &'a mut Encoder>,
-    {
+    pub(crate) fn apply<'a>(&'a self, encoders: &'a mut[Encoder], scene: &'a Scene) -> DrawUpdate<'a> {
         use num_cpus;
 
         self.clear_color.map(|c| {
-            self.target.clear_color(encoders.nth(0).unwrap(), c)
+            self.target.clear_color(&mut encoders[0], c)
         });
         self.clear_depth.map(|d| {
-            self.target.clear_depth_stencil(encoders.nth(0).unwrap(), d)
+            self.target.clear_depth_stencil(&mut encoders[0], d)
         });
 
-        let update = self.passes
-            .iter()
-            .map(|pass| {
-                let mut models = scene.par_chunks_models(num_cpus::get());
-                let enc = encoders.by_ref().take(models.len()).collect::<Vec<_>>();
-                models.zip(enc).map_with(
-                    pass,
-                    (|pass, (models, enc)| (*pass, models, enc)) as ApplyPassFn<'a>,
-                )
-            })
-            .collect::<Vec<_>>();
+        let mut encoders = encoders.iter_mut();
+        let mut update = Vec::new();
+        for pass in self.passes.iter() {
+            let mut models = scene.par_chunks_models(num_cpus::get());
```

Comment:
I suggest to use `encoders.len() / passes.len()`.
Note that `Stage::encoders_required` doesn't use `num_cpu::get()` also.